fireworks_poe_bot/fw_poe_server_bot.py

```diff
@@ -21,6 +21,50 @@
 import logging
 import time
 import os
+import io
+import base64
+import httpx
+from PIL import Image
+import math
+
+
+async def download_image_and_encode_to_base64(
+    url: str,
+) -> Tuple[str, Dict[str, Union[int, str]]]:
+    async with httpx.AsyncClient() as client:
+        image_download_start = time.perf_counter()
+        r = await client.get(url)
+        image_download_end = time.perf_counter()
+        if r.status_code == 200:
+            img_buffer = r.content
+            img_resize_factor = 1
+            encode_start = time.perf_counter()
+            if len(r.content) / 1024 / 1024 > 1.5:
+                # if the image is bigger than 1.5 MB, then we downsize the image
+                # downscale so that the image size is smaller than 1.5 MB
+                img_resize_factor = math.ceil(
+                    math.sqrt(len(r.content) / 1024 / 1024 / 1.5)
+                )
+                pil_img = Image.open(io.BytesIO(r.content))
+                pil_img_resized = pil_img.resize(
+                    (x // img_resize_factor for x in pil_img.size)
+                )
+                buffered = io.BytesIO()
+                pil_img_resized.save(buffered, format="JPEG")
+                img_buffer = buffered.getvalue()
+            img = "data:image/jpeg;base64,{}".format(
+                base64.b64encode(img_buffer).decode("utf-8")
+            )
```

Comment:
Discussed offline: just normalize the size of this image here instead of resizing based on a ratio